File: sdk/typescript/src/___PLAYGROUND___/relayer/clone-relayer.ts
import { begin } from '../helpers';

export const cloneRelayer = async () => {
  const context = await begin();

  console.log('Creating new relayer...');
  const relayer = await context.client.relayer.clone(
    '94afb207-bb47-4392-9229-ba87e4d783cb',
    31337,
    `test-relayer-${Date.now()}`
  );
  console.log('Created relayer:', relayer);

  // Clean up - delete the test relayer
  await context.client.relayer.delete(relayer.id);
  console.log('Test relayer cleaned up');

  await context.end();
};

cloneRelayer().then(() => console.log('clone-relayer done'));

File: sdk/python/src/__PLAYGROUND__/relayer/clone-relayer.py
import asyncio
from datetime import datetime

from __PLAYGROUND__.helper import begin, end


async def cloneRelayer():
    rrelayer_node = None

    try:
        client, _, info, rrelayer_node = await begin()

        print("Creating new relayer...")
        relayer = await client.relayer.clone(
            info["id"], 31337, f"test-relayer-{datetime.now()}"
        )

        print("Created relayer", relayer)

        await client.relayer.delete(relayer["id"])
        print("Test relayer cleaned up")
    except Exception as e:
        print("cloneRelayer failed:", e)
    finally:
        if rrelayer_node is not None:
            end(rrelayer_node)


if __name__ == "__main__":
    asyncio.run(cloneRelayer())
    print("clone-relayer done")
